lib/elixir/src/elixir_errors.erl

```diff
@@ -57,6 +61,119 @@ emit_diagnostic(Severity, Position, File, Message, Stacktrace) ->
 
   ok.
 
+% --------- Fancy Diagnostics
+
+fancy_exception_snippet(File, LineNumber, Column, Description, Snippet) ->
+  #{content := Content, offset := Offset} = Snippet,
+  {TrimmedContent, TotalLeading} = trim_line(Content),
+  LineDigits = get_line_number_digits(LineNumber),
+  Spacing = n_spaces(LineDigits + 1),
+
+  Formatted = io_lib:format(
+    " ~ts┌─ ~ts~ts\n"
+    " ~ts│\n"
+    " ~p │ ~ts\n"
+    " ~ts│ ~ts\n"
+    " ~ts│\n"
+    " ~ts~ts",
+    [
+     Spacing, prefix(error), file_format({LineNumber, Column}, File),
+     Spacing,
+     LineNumber, TrimmedContent,
+     Spacing, highlight_below_line((Offset - TotalLeading) + 1, error),
+     Spacing,
+     Spacing, format_message(Description, LineDigits)
+    ]),
+
+  unicode:characters_to_binary(Formatted).
+
+fancy_exception(File, LineNumber, Column, Message, ShowLine) ->
+  Formatted = case filelib:is_regular(File) andalso ShowLine of 
+             true -> 
+               line_column_diagnostic({LineNumber, Column}, File, Message, error);
```

Comment:
@josevalim do you think it makes sense for us to try and get the line when there's no snippet?  Many examples I've seen contain wrong line information (such as in token missing), so we don't fetch them. 

Do you have any suggestions on some piece of code we can use to test this (where the compiler will crash but will have the correct file information to read it (without providing `file`)? Currently we're only testing cases with snippets

Reply:
> @josevalim do you think it makes sense for us to try and get the line when there's no snippet? Many examples I've seen contain wrong line information (such as in token missing), so we don't fetch them.
> 
> Do you have any suggestions on some piece of code we can use to test this (where the compiler will crash but will have the correct file information to read it, so `@file` wouldnt work)? Currently we're only testing cases with snippets